### PR TITLE
feat: load skills by source precedence

### LINKED ISSUE #400: feat: skills 支持项目级发现、注册与优先级解析
Split from #396.

## 背景
当前 skills 只覆盖 `builtin` 和 `app`；`get_project_skills_dir()` 实际返回 app 目录，repo-local skills 不会进入 runtime。OpenSpec 安装后的项目级 skills 因此不可见。

## 目标
让 relay-teams 支持项目级 skill 发现、注册和优先级解析，使仓库内 skills 无需安装到 `~/.relay-teams/skills` 就能被运行时直接发现和加载。

## 实现范围
- 新增 `SkillScope.PROJECT`。
- 技能发现改为三层作用域：`builtin < app < project`。
- 项目级 roots 明确为：
  - `<workspace_root>/.relay-teams/skills`
  - 兼容 `<workspace_root>/.codex/skills` 作为 project scope 的补充来源
- 同名 skill 按 plain name 解析时，优先级改为 `project > app > builtin`。
- canonical ref 继续保留全部条目，例如 `project:xxx`、`app:xxx`、`builtin:xxx`。
- 更新：
  - `SkillsDirectory` 来源遍历逻辑
  - `SkillRegistry.from_default_scopes()`
  - CLI `skills list/show`
  - system prompt 中的 available skills 列表
  - `reload_skills_config` 对 project scope 的刷新能力

## 技术要求
- 继续遵守现有校验分层：
  - 显式用户配置时，未知 skill 仍严格报错
  - 运行时遇到 persisted 脏数据时，过滤缺失 skill 并记录 warning，不得导致启动或执行崩溃
- CLI `--source` 需要新增 `project`。
- `load_skill` 的说明文案和歧义处理更新为 project-first。
- 必须遵守当前接口边界，不能只改 CLI；后端 registry / service / prompt 侧都要对齐。

## 验收标准
- 仓库内 skill 无需安装到 app 目录即可被列出并加载。
- 同名 skill 同时存在 project / app / builtin 时，plain name 命中 project 版本。
- canonical ref 仍可显式访问不同 scope 的同名 skill。
- `reload_skills_config` 后，新增或删除的 project skill 能反映到运行时。

## 测试要求
- unit：scope 解析、优先级、重复命名、fallback root。
- integration：在临时 workspace 放入 repo-local skill，run prompt 时能看到并 `load_skill` 成功。
- CLI：`skills list --source project`、`skills show project:...`。

## 非目标
- 不新增 skill 执行引擎。
- 不做 OpenSpec 专用 skill 解析逻辑，只补项目级 runtime 能力。

### PR DESCRIPTION
**Pull request by yexxx**
## Summary
- replace builtin/app-specific skill identity handling with source-ordered discovery and bare skill refs
- load skills in `builtin -> ~/.relay-teams -> ~/.agents -> project .relay-teams -> project .agents` order, with later sources overriding earlier duplicate names
- align CLI, server/runtime, ClawHub, docs, and frontend consumers with the new `source` model and add broader edge-case unit coverage

## Validation
- `uv run --extra dev pytest -q tests/unit_tests/skills tests/unit_tests/roles tests/unit_tests/interfaces tests/unit_tests/providers tests/unit_tests/frontend`
- `uv run --extra dev ruff check src tests`
- `uv run --extra dev basedpyright`
- real-world load check with OpenSpec-generated skills copied into project `.agents/skills`

Closes #400

### CODE REVIEW

File: src/relay_teams/skills/discovery.py [hunk 1]
```diff
@@ -254,13 +270,72 @@ def _load_skill(self, *, path: Path, scope: SkillScope) -> Skill | None:
...
+def _project_skill_sources(*, start_dir: Path) -> tuple[tuple[SkillSource, Path], ...]:
+    resolved_start_dir = _resolve_start_dir(start_dir)
+    project_root = get_project_root_or_none(start_dir=resolved_start_dir)
+    stop_dir = resolved_start_dir if project_root is None else project_root
+    sources: list[tuple[SkillSource, Path]] = []
+    for current_dir in _iter_parent_dirs(resolved_start_dir, stop_dir):
+        sources.append(
+            (
+                SkillSource.PROJECT_RELAY_TEAMS,
+                current_dir / ".relay-teams" / "skills",
```
[the remark this answers is not in the document]
Comment (yexxx) [reply src/relay_teams/skills/discovery.py]: Fixed in `83e36bf6`.

Project skill sources are now built in source-precedence groups instead of being interleaved by directory depth, so the effective order remains:
`builtin -> ~/.relay-teams -> ~/.agents -> project .relay-teams -> project .agents`.

I also added a regression test that specifically covers a nested `.agents/skills` override versus a parent `.relay-teams/skills` duplicate.

File: src/relay_teams/skills/skill_registry.py [hunk 2]
```diff
@@ -363,59 +319,19 @@ async def _action() -> JsonValue:
...
+    ) -> tuple[tuple[str, ...], tuple[str, ...]]:
         skill_map = self._get_effective_skill_map()
-        skill_name_index = self._get_skill_name_index(skill_map)
         resolved_names: list[str] = []
         missing_names: list[str] = []
-        ambiguous_names: dict[str, tuple[str, ...]] = {}
         for raw_name in skill_names:
             name = raw_name.strip()
+            if not name:
+                continue
```
[the remark this answers is not in the document]
Comment (yexxx) [reply src/relay_teams/skills/skill_registry.py:346]: Fixed in `83e36bf6`.

Strict resolution now treats blank entries as invalid input again, while lenient runtime recovery paths still ignore them. Added regression coverage for both `resolve_known(..., strict=True)` and `validate_known(...)` with blank skill names.

File: src/relay_teams/interfaces/server/routers/roles.py [hunk 3]
```diff
@@ -216,32 +216,31 @@ def _load_role_skill_options(
...
-def _collect_required_builtin_skill_refs(role_registry: RoleRegistry) -> frozenset[str]:
+def _collect_required_builtin_skill_names(
+    role_registry: RoleRegistry,
+) -> frozenset[str]:
     return frozenset(
-        skill_name
-        for role in role_registry.list_roles()
-        for skill_name in role.skills
-        if skill_name.startswith("builtin:")
+        skill_name for role in role_registry.list_roles() for skill_name in role.skills
```
[the remark this answers is not in the document]
Comment (yexxx) [reply src/relay_teams/interfaces/server/routers/roles.py]: Fixed in `3a0737f4`.

`/api/roles:options` now only treats skills referenced by reserved system roles as required for the builtin-skill reload/availability check, so missing user/project skill drift no longer escalates into a 503. Added a regression test covering a user role that still references a removed project skill.

File: src/relay_teams/interfaces/server/container.py [hunk 4]
```diff
@@ -281,7 +281,8 @@ def __init__(
         self.mcp_registry: McpRegistry = self.mcp_config_manager.load_registry()
         self.mcp_service: McpService = McpService(registry=self.mcp_registry)
         self.skill_registry: SkillRegistry = SkillRegistry.from_config_dirs(
-            app_config_dir=app_config_dir
+            app_config_dir=app_config_dir,
+            project_start_dir=Path.cwd(),
```
[the remark this answers is not in the document]
Comment (yexxx) [reply src/relay_teams/interfaces/server/container.py]: Fixed in `cc1094c9`.

The server container no longer seeds its shared `SkillRegistry` with `Path.cwd()`, so startup location no longer leaks one project's overrides into another project's server session. I also added a container regression test that asserts `from_config_dirs(...)` is called without a `project_start_dir`.

File: src/relay_teams/interfaces/server/runtime_identity.py [hunk 5]
```diff
@@ -107,20 +106,25 @@ def build_skill_registry_sanity(
         if config_dir is None
         else config_dir.expanduser().resolve()
     )
-    registry = (
+    _ = (
         skill_registry
         if skill_registry is not None
-        else SkillRegistry.from_config_dirs(app_config_dir=resolved_config_dir)
+        else SkillRegistry.from_config_dirs(
+            app_config_dir=resolved_config_dir,
```
[the remark this answers is not in the document]
Comment (yexxx) [reply src/relay_teams/interfaces/server/runtime_identity.py]: Fixed in `cc1094c9`.

`build_skill_registry_sanity(...)` now projects builtin skill health from the injected active registry when one is provided, instead of reconstructing a separate builtin-only registry and hiding runtime drift. Added a regression test that proves a user override changes the reported builtin sanity view.

File: src/relay_teams/skills/config_reload_service.py [hunk 6]
```diff
@@ -32,7 +32,8 @@ def reload_skills_config(self) -> SkillRegistry:
             attributes={"config_dir": str(self._config_dir)},
         ):
             skill_registry = SkillRegistry.from_config_dirs(
-                app_config_dir=self._config_dir
+                app_config_dir=self._config_dir,
+                project_start_dir=Path.cwd(),
             )
```
[the remark this answers is not in the document]
Comment (yexxx) [reply src/relay_teams/skills/config_reload_service.py]: Fixed in `3a3fcf28`.

`SkillsConfigReloadService.reload_skills_config()` now rebuilds the registry from `app_config_dir` only, so reload paths no longer depend on the daemon launch directory or pull project `.relay-teams/.agents` overrides into the shared runtime state. I also added a regression test that asserts reload calls `SkillRegistry.from_config_dirs(...)` without any `project_start_dir`.

File: src/relay_teams/skills/discovery.py [hunk 7]
```diff
@@ -97,10 +101,15 @@ def from_default_scopes(
...
     ) -> SkillsDirectory:
-        return cls.from_skill_dirs(
-            app_skills_dir=get_app_skills_dir(user_home_dir=user_home_dir),
-            builtin_skills_dir=get_builtin_skills_dir_path(),
+        return cls(
+            sources=_build_default_sources(
+                builtin_skills_dir=get_builtin_skills_dir_path(),
+                relay_teams_skills_dir=get_app_skills_dir(user_home_dir=user_home_dir),
+                agents_skills_dir=get_agents_skills_dir(user_home_dir=user_home_dir),
+                project_start_dir=start_dir if start_dir is not None else Path.cwd(),
```
[the remark this answers is not in the document]
Comment (yexxx) [reply src/relay_teams/skills/discovery.py]: Fixed in `dfc1e287`.

`SkillsDirectory.from_default_scopes()` now only includes project `.relay-teams/.agents` sources when callers explicitly pass `start_dir`; the default path is back to builtin + user scopes only. That keeps installer/support flows like `mount_skills_to_roles()` from treating project-only skills as globally bindable, and I added regression coverage for both the default-source construction and the installer rejection path.

File: src/relay_teams/interfaces/server/container.py [hunk 8]
```diff
@@ -281,7 +281,7 @@ def __init__(
         self.mcp_registry: McpRegistry = self.mcp_config_manager.load_registry()
         self.mcp_service: McpService = McpService(registry=self.mcp_registry)
         self.skill_registry: SkillRegistry = SkillRegistry.from_config_dirs(
-            app_config_dir=app_config_dir
+            app_config_dir=app_config_dir,
```
[the remark this answers is not in the document]
Comment (yexxx) [reply src/relay_teams/interfaces/server/container.py:297]: Fixed in `f2c1aa60`.

The server container now captures a single explicit project root snapshot at startup and passes it into both the initial `SkillRegistry.from_config_dirs(...)` call and every subsequent `SkillsConfigReloadService` rebuild. That restores project `.relay-teams/.agents` skill visibility for server/runtime paths while keeping reload behavior stable instead of re-reading the live process cwd. I also updated regression coverage for both the container initialization call and reload-service propagation.

File: frontend/dist/js/components/settings/index.js [hunk 9]
```diff
@@ -349,14 +349,14 @@ function createModal() {
                                             </div>
                                         </div>
                                         <div class="profile-editor-subsection">
-                                            <h5 data-i18n="settings.model.fallback_section">Fallback</h5>
-                                            <div class="form-row profile-fallback-row">
+                                            <h5>Fallback</h5>
+                                            <div class="form-row">
                                                 <div class="form-group">
-                                                    <label for="profile-fallback-policy" data-i18n="settings.model.fallback_strategy">Fallback Strategy</label>
+                                                    <label for="profile-fallback-policy">Fallback Strategy</label>
```
[the remark this answers is not in the document]
Comment (yexxx) [reply frontend/dist/js/components/settings/index.js]: Fixed in 903a4b5e.

Restored data-i18n bindings for the fallback subsection labels and added the corresponding settings.model.fallback* translation keys in both en-US and zh-CN, so the model fallback controls localize correctly again.

File: frontend/dist/js/components/settings/modelProfiles.js [hunk 10]
```diff
@@ -1545,7 +1523,7 @@ function renderProfileCard(name, profile, index) {
     const baseUrlLabel = profile.base_url || t('settings.model.no_endpoint');
     const fallbackLabel = profile.fallback_policy_id
         ? formatFallbackPolicyLabel(profile.fallback_policy_id)
-        : t('settings.model.fallback_disabled');
+        : 'Fallback disabled';
```
[the remark this answers is not in the document]
Comment (yexxx) [reply frontend/dist/js/components/settings/modelProfiles.js]: Fixed in 40c5425.

Localized the fallback profile-card summary and the fallback-policy select default option by routing them through 	(...)/ormatMessage(...), and added the corresponding settings.model.* translation keys for both en-US and zh-CN.

File: src/relay_teams/skills/skill_registry.py [hunk 11]
```diff
@@ -364,58 +323,22 @@ def _resolve_skill_names(
...
-                chosen_skill = candidates[0]
-                resolved_names.append(chosen_skill.ref)
-                self._log_ambiguous_skill_name_resolution(
-                    requested_name=name,
-                    chosen_ref=chosen_skill.ref,
-                    candidate_refs=candidate_refs,
-                    consumer=consumer,
-                )
             else:
                 missing_names.append(name)
```
[the remark this answers is not in the document]
Comment (yexxx) [reply src/relay_teams/skills/skill_registry.py]: Fixed in 037af5f8.

Strict resolution now normalizes legacy builtin:/app: skill refs to bare names when the effective skill exists, while preserving the original scoped string for true unknown-skill errors. Added regression coverage for resolve_known(...) and validate_known(...).

File: frontend/dist/js/components/settings/modelProfiles.js [hunk 12]
```diff
@@ -144,10 +144,6 @@ export function bindModelProfileHandlers() {
     if (imageCapabilityInput) {
         imageCapabilityInput.onchange = syncDraftImageCapabilityMode;
     }
-
-    if (typeof document?.addEventListener === 'function') {
-        document.addEventListener('agent-teams-language-changed', handleModelProfileLanguageChanged);
-    }
 }
```
[the remark this answers is not in the document]
Comment (yexxx) [reply frontend/dist/js/components/settings/modelProfiles.js:157]: Fixed in 037af5f8.

Model profiles now bind agent-teams-language-changed again. The handler rerenders the list or editor dynamic content as appropriate, and fallback-policy options preserve the current selection while their labels are refreshed. Added frontend regression coverage for both list-card fallback copy and the editor fallback select.

File: frontend/dist/js/components/settings/modelProfiles.js [hunk 13]
```diff
@@ -1747,32 +1751,26 @@ function renderFallbackPolicyOptions() {
...
     }
-    const selectedValue = String(select.value || '');
+    const selectedValue = String(select.value || '').trim();
     const policies = Array.isArray(fallbackConfig?.policies) ? fallbackConfig.policies : [];
     const options = [
-        `<option value="">${escapeHtml(t('settings.model.fallback_disabled_option'))}</option>`,
+        `<option value="">${escapeHtml(t('settings.model.disabled'))}</option>`,
         ...policies.map(policy => (
-            `<option value="${escapeHtml(policy.policy_id)}">${escapeHtml(formatFallbackPolicyLabel(policy.policy_id, policy.name))}</option>`
+            `<option value="${escapeHtml(policy.policy_id)}">${escapeHtml(policy.name || policy.policy_id)}</option>`
```
[the remark this answers is not in the document]
Comment (yexxx) [reply frontend/dist/js/components/settings/modelProfiles.js]: Fixed in ed658f39.

Built-in fallback policies now resolve through locale-aware translation keys instead of echoing the backend English names. Unknown/custom policies still fall back to the server-provided display name. Added frontend regression coverage for both profile-card summaries and the editor dropdown on language changes.

**Review by stevetdp — APPROVED** [review]
